source/extensions/filters/network/zookeeper_proxy/decoder.cc

```diff
@@ -204,7 +210,9 @@ void DecoderImpl::decodeOnWrite(Buffer::Instance& data, uint64_t& offset) {
 
   // Control responses that aren't connect, with XIDs <= 0.
   const auto zxid = helper_.peekInt64(data, offset);
+  ENVOY_LOG(trace, "zookeeper_proxy: decoding response with zxid {} at offset {}", zxid, offset);
   const auto error = helper_.peekInt32(data, offset);
+  ENVOY_LOG(trace, "zookeeper_proxy: decoding response with error {} at offset {}", error, offset);
```

Comment:
Same here. We don't need 5 lines of logs.

Reply:
I am thinking if we merge all logs in the end and some errors happen above, we may lost all the information.

Reply:
IMO we can merge 172, 178 into 182. and merge 213 and 215. 
Any other concerns?

Reply:
So if we merge L172, L177 into L182 and either `ensureMinLength` or `ensureMaxLength` raises an exception, we only know the packet is either too big or too small but not know the exact length extracted from the buffer.

Reply:
Thanks for your clarification!
Log will show the exception body IIRC, but I agree that we can keep log before those exception.
On the other hand, we should avoid exception in the data path if possible, but that could be a follow-up.

Let's merge the logs as much as possible but still keep the info exposure at the same degree.